File: test/support/live_state/live-state.js
export function init() {
  return {
    customers: [
      {
        firstName: "bob",
        lastName: "jones",
        email: "bob@jones.com"
      }
    ],
    editingCustomer: {
      firstName: "bob",
      lastName: "notbob",
      email: "bob@notbob.com"
    }
  };
}

export function addCustomer(customer, state) {
  return state;
}

File: test/support/live_state/live_state.ex
defmodule WasmComponentsEx.Test.LiveState do
  alias WasmComponentsEx.Test.LiveState.Native

  defstruct instance: nil, store: nil, reference: nil

  def init(%__MODULE__{instance: instance, store: store}), do: Native.init(store, instance)

  def add_customer(%__MODULE__{instance: instance, store: store}, customer, state), do: Native.add_todo(store, instance, customer, state)

  def new(component_bytes, options \\ []) do
    store = Native.new_store(%WasmComponentsEx.WasiOptions{}, %WasmComponentsEx.StoreLimits{})
    component = Native.new_component(store, component_bytes)
    instance = Native.instantiate(store, component)
    {:ok, %__MODULE__{instance: instance, store: store}}
  end

end
